ibis/bigquery/compiler.py

```diff
@@ -158,23 +158,20 @@ def approx_quantile(arg, number, distinct=False, ignore_nulls=True):
     return bq_ops.ApproxQuantile(arg, number, distinct, ignore_nulls).to_expr()
 
 
-def _approx_count_distinct(translator, expr):
+def _approx_nunique(translator, expr):
     arg = expr.op().args[0]
     arg_formatted = translator.translate(arg)
     return 'APPROX_COUNT_DISTINCT({})'.format(arg_formatted)
 
 
-def approx_count_distinct(arg):
-    return bq_ops.ApproxCountDistinct(arg).to_expr()
-
-
 def _format_date(translator, expr):
     (arg, fmt) = map(translator.translate, expr.op().args)
-    return 'FORMAT_DATE({}, {})'.format(fmt, arg)
-
-
-def format_date(fmt, arg):
-    return bq_ops.FormatDate(arg, fmt).to_expr()
+    klass_name = expr.op().args[0].type().__class__.__name__.upper()
+    if klass_name in ['TIMESTAMP', 'DATE', 'TIME']:
```

Comment:
This should be an `assert`ion since you shouldn't be able to reach this code path if the type system is working correctly.